src/magnet_coils.py

```diff
@@ -0,0 +1,582 @@
+import log
+import cubit
+import numpy as np
+from pathlib import Path
+import subprocess
+import yaml
+import argparse
+
+m2cm = 100
+
+def unit_vector(vec):
+    """Normalizes given vector.
+
+    Arguments:
+        vec (array of [float, float, float]): input vector to be normalized.
+
+    Returns:
+        vec (array of [float, float, float]): normalized input vector.
+    """
+
+    vec = vec/((np.sum(vec**2))**0.5)
```

Comment:
`np.squared` and `np.sqrt` may be safer.  Have we tested that this works with an array of vectors?

Reply:
`np.square` definitely would be a good call. np.sum() should always return a scalar in our use case (normalizing 1D arrays) so I would imagine the **0.5 should be fine but maybe it would be best to just use np.sqrt anyway? I could add a check as well to make sure the results are as expected? Either in the function itself or as a test?

Reply:
I don't think we need to add a test here, but it would be good to test this in a notebook just to make sure it works in the case with an Numpy array of N vectors - kind of like we were doing on Saturday.

Reply:
Also, I always have the sense that `sqrt()` functions are more robust that `**0.5`, and the `np.sum` will be a vector of scalars if `vec` is effectively a list of vectors.

Reply:
There is a version of such a function in #55 and we should probably pick one and be confident it will work in all our use cases and make it available to all the methods. (ping @connoramoreno )

Reply:
I've added a `utils.py` utility script to the `src` directory that contains the `normalize()` function I use in #55, though that `normalize()` function only works on 2-D arrays. I can change it to work for arrays of any dimension but it might be better just to use the `normalize()` function from the `scikit-learn.preprocessing` Python module, as that has all the functionality we'd need.

Reply:
my first instinct was to avoid another dependency for just this functionality, but I'm happy to hear arguments about one vs the other vis-a-vis robustness

Reply:
Well we know for certain that `sklearn.preprocessing.normalize()` is robust and can handle arrays of any dimension, with some caveats for 1-D arrays (though maybe that isn't important for our use case). I suppose `src.utils.normalize()` as-is can also handle ndarrays but not 1-D arrays (again, may not be important for our use case)

Reply:
`np.linalg.norm` might be another candidate? Would need to handle 1D arrays specially (axis 1 doesnt exist) but can be done with numpy. Example:

```
array([[1, 2, 3],
       [4, 5, 6],
       [7, 8, 9]])
>>> norm_vec = np.linalg.norm(arr,axis=1)
>>> norm_vec
array([ 3.74165739,  8.77496439, 13.92838828])
>>> normalized_arr = np.divide(arr.T, norm_vec).T
>>> normalized_arr
array([[0.26726124, 0.53452248, 0.80178373],
       [0.45584231, 0.56980288, 0.68376346],
       [0.50257071, 0.57436653, 0.64616234]])
>>> np.linalg.norm(normalized_arr, axis=1)
array([1., 1., 1.])

>>> arr = np.array([1,2,3])
>>> norm_vec = np.linalg.norm(arr, axis=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/filespace/e/epflug/research/mambaforge/lib/python3.10/site-packages/numpy/linalg/linalg.py", line 2583, in norm
    return sqrt(add.reduce(s, axis=axis, keepdims=keepdims))
numpy.exceptions.AxisError: axis 1 is out of bounds for array of dimension 1
>>> 


```